### ISSUE TITLE
在ubuntu16.4系统的Anaconda虚拟环境中安装jittor cuda启用失败

### ISSUE THREAD

**Issue by 123world** [Jittor/jittor#214]
安装的系统是ubuntu16.4系统，安装了g++，创建的虚拟环境python版本3.7
cuda和cudnn都已经安装

(jt2) super@super-SYS-4028GR-TR:/$ python -m jittor.test.test_example
[i 0513 13:48:48.375195 00 compiler.py:851] Jittor(1.2.2.73) src: /home/super/anaconda3/envs/jt2/lib/python3.7/site-packages/jittor
[i 0513 13:48:48.375360 00 compiler.py:852] g++ at /usr/bin/g++
[i 0513 13:48:48.375468 00 compiler.py:853] cache_path: /home/super/.cache/jittor/default/g++
[i 0513 13:48:48.384136 00 __init__.py:255] Found /usr/local/cuda/bin/nvcc(9.2.148) at /usr/local/cuda/bin/nvcc.
[i 0513 13:48:48.467075 00 __init__.py:255] Found gdb(7.11.1) at /usr/bin/gdb.
[i 0513 13:48:48.476718 00 __init__.py:255] Found addr2line(2.26.1) at /usr/bin/addr2line.
[i 0513 13:48:48.499283 00 compiler.py:893] py_include: -I/home/super/anaconda3/envs/jt2/include/python3.7m -I/home/super/anaconda3/envs/jt2/include/python3.7m
[i 0513 13:48:48.520513 00 compiler.py:895] extension_suffix: .cpython-37m-x86_64-linux-gnu.so
[i 0513 13:48:59.897264 64 __init__.py:167] Total mem: 125.80GB, using 16 procs for compiling.
Compiling jittor_core(143/143) used: 25.674s eta: 0.000s
[i 0513 13:49:25.985628 64 jit_compiler.cc:21] Load cc_path: /usr/bin/g++
[i 0513 13:49:25.985787 64 init.cc:54] Found cuda archs: []
gcc: error: unrecognized command line option ‘--showme:version’
[i 0513 13:49:26.170539 64 compile_extern.py:392] mpicc not found, distribution disabled.
Compiling gen_ops_mkl_conv_backward_x_mkl_test_mkl_matmul_mkl_conv_mkl_conv_backCompiling gen_ops_mkl_conv_backward_x_mkl_test_mkl_matmul_mkl_conv_mkl_conv_backward_w(7/7) used: 2.924s eta: 0.000s
Compiling gen_ops_cub_argsort_cub_arg_reduce_cub_test_cub_where(5/5) used: 2.124s eta: 0.000s
[i 0513 13:49:34.852806 64 compile_extern.py:18] found /usr/local/cuda/include/cublas.h
[i 0513 13:49:34.858197 64 compile_extern.py:18] found /usr/local/cuda/lib64/libcublas.so
[i 0513 13:49:36.584562 64 compile_extern.py:18] found /usr/local/cuda/include/cudnn.h
[i 0513 13:49:36.591098 64 compile_extern.py:18] found /usr/local/cuda/lib64/libcudnn.so
[i 0513 13:49:36.608752 64 compiler.py:658] handle pyjt_include/home/super/anaconda3/envs/jt2/lib/python3.7/site-packages/jittor/extern/cuda/cudnn/inc/cudnn_warper.h
Compiling gen_ops_cudnn_conv_backward_x_cudnn_test_cudnn_conv_cudnn_conv_backward_w(9/9) used: 2.588s eta: 0.000s
[i 0513 13:49:39.286657 64 compile_extern.py:18] found /usr/local/cuda/include/curand.h
[i 0513 13:49:39.292560 64 compile_extern.py:18] found /usr/local/cuda/lib64/libcurand.so

Compiling Operators(10/10) used: 2.31s eta:    0s 
step 0, loss = 0.8328441381454468 {'hold_vars': 14, 'lived_vars': 64, 'lived_ops': 57}

Compiling Operators(9/9) used: 3.32s eta:    0s 
step 1, loss = 0.5198176503181458 {'hold_vars': 14, 'lived_vars': 64, 'lived_ops': 57}

出现的问题好象是
[i 0513 13:49:25.985787 64 init.cc:54] Found cuda archs: []
gcc: error: unrecognized command line option ‘--showme:version’
[i 0513 13:49:26.170539 64 compile_extern.py:392] mpicc not found, distribution disabled.
但是不知道怎么解决，网上搜索也没有找到，请问这有解决方案吗

**Comment by Jittor** [comment]
谢谢反馈，有可能是您的cuda版本太老了（9.2），您nvidia-smi输出是怎样的呢？



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2021年5月13日(周四) 下午2:14
收件人: ***@***.***&gt;;
抄送: ***@***.***&gt;;
主题: [Jittor/jittor] 在ubuntu16.4系统的Anaconda虚拟环境中安装jittor cuda启用失败 (#214)




 
安装的系统是ubuntu16.4系统，安装了g++，创建的虚拟环境python版本3.7
 cuda和cudnn都已经安装
 
(jt2) ***@***.***SYS-4028GR-TR:/$ python -m jittor.test.test_example
 [i 0513 13:48:48.375195 00 compiler.py:851] Jittor(1.2.2.73) src: /home/super/anaconda3/envs/jt2/lib/python3.7/site-packages/jittor
 [i 0513 13:48:48.375360 00 compiler.py:852] g++ at /usr/bin/g++
 [i 0513 13:48:48.375468 00 compiler.py:853] cache_path: /home/super/.cache/jittor/default/g++
 [i 0513 13:48:48.384136 00 init.py:255] Found /usr/local/cuda/bin/nvcc(9.2.148) at /usr/local/cuda/bin/nvcc.
 [i 0513 13:48:48.467075 00 init.py:255] Found gdb(7.11.1) at /usr/bin/gdb.
 [i 0513 13:48:48.476718 00 init.py:255] Found addr2line(2.26.1) at /usr/bin/addr2line.
 [i 0513 13:48:48.499283 00 compiler.py:893] py_include: -I/home/super/anaconda3/envs/jt2/include/python3.7m -I/home/super/anaconda3/envs/jt2/include/python3.7m
 [i 0513 13:48:48.520513 00 compiler.py:895] extension_suffix: .cpython-37m-x86_64-linux-gnu.so
 [i 0513 13:48:59.897264 64 init.py:167] Total mem: 125.80GB, using 16 procs for compiling.
 Compiling jittor_core(143/143) used: 25.674s eta: 0.000s
 [i 0513 13:49:25.985628 64 jit_compiler.cc:21] Load cc_path: /usr/bin/g++
 [i 0513 13:49:25.985787 64 init.cc:54] Found cuda archs: []
 gcc: error: unrecognized command line option ‘--showme:version’
 [i 0513 13:49:26.170539 64 compile_extern.py:392] mpicc not found, distribution disabled.
 Compiling gen_ops_mkl_conv_backward_x_mkl_test_mkl_matmul_mkl_conv_mkl_conv_backCompiling gen_ops_mkl_conv_backward_x_mkl_test_mkl_matmul_mkl_conv_mkl_conv_backward_w(7/7) used: 2.924s eta: 0.000s
 Compiling gen_ops_cub_argsort_cub_arg_reduce_cub_test_cub_where(5/5) used: 2.124s eta: 0.000s
 [i 0513 13:49:34.852806 64 compile_extern.py:18] found /usr/local/cuda/include/cublas.h
 [i 0513 13:49:34.858197 64 compile_extern.py:18] found /usr/local/cuda/lib64/libcublas.so
 [i 0513 13:49:36.584562 64 compile_extern.py:18] found /usr/local/cuda/include/cudnn.h
 [i 0513 13:49:36.591098 64 compile_extern.py:18] found /usr/local/cuda/lib64/libcudnn.so
 [i 0513 13:49:36.608752 64 compiler.py:658] handle pyjt_include/home/super/anaconda3/envs/jt2/lib/python3.7/site-packages/jittor/extern/cuda/cudnn/inc/cudnn_warper.h
 Compiling gen_ops_cudnn_conv_backward_x_cudnn_test_cudnn_conv_cudnn_conv_backward_w(9/9) used: 2.588s eta: 0.000s
 [i 0513 13:49:39.286657 64 compile_extern.py:18] found /usr/local/cuda/include/curand.h
 [i 0513 13:49:39.292560 64 compile_extern.py:18] found /usr/local/cuda/lib64/libcurand.so
 
Compiling Operators(10/10) used: 2.31s eta:    0s
 step 0, loss = 0.8328441381454468 {'hold_vars': 14, 'lived_vars': 64, 'lived_ops': 57}
 
Compiling Operators(9/9) used: 3.32s eta:    0s
 step 1, loss = 0.5198176503181458 {'hold_vars': 14, 'lived_vars': 64, 'lived_ops': 57}
 
出现的问题好象是
 [i 0513 13:49:25.985787 64 init.cc:54] Found cuda archs: []
 gcc: error: unrecognized command line option ‘--showme:version’
 [i 0513 13:49:26.170539 64 compile_extern.py:392] mpicc not found, distribution disabled.
 但是不知道怎么解决，网上搜索也没有找到，请问这有解决方案吗

**Comment by 123world** [comment]
驱动版本396.37

**Comment by maxint** [comment]
@123world 这里显示使用 cuda 版本太老
```
[i 0513 13:48:48.384136 00 init.py:255] Found /usr/local/cuda/bin/nvcc(9.2.148) at /usr/local/cuda/bin/nvcc.
```
可以通过修改LD_LIBRARY_PATH和PATH替换版本
```
export CUDA_HOME=/usr/local/cuda-11.1 # 10.0
export LD_LIBRARY_PATH=$CUDA_HOME/lib64 # Use CUDA 10.0 to support TitanXp
export PATH=$CUDA_HOME/bin:$PATH
```

**Comment by wzr0108** [comment]
请问你解决了吗，我也遇到了这个问题gcc: error: unrecognized command line option ‘--showme:version’
